nautobot/extras/models/groups.py

```diff
@@ -853,6 +864,53 @@ def get_group_members_url(self):
         """Return the group members URL."""
         return self.group.get_group_members_url()
 
+    def get_siblings(self, include_self=False):
+        """Return group memberships that share the same parent group."""
+        siblings = DynamicGroupMembership.objects.filter(parent_group=self.parent_group)
+        if include_self:
+            return siblings
+        return siblings.exclude(pk=self.pk)
```

Comment:
Any particular reason we're using `pk` in this `get_siblings` but using `slug` in the other `get_siblings` above? Could/should we use pk for both?